src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -59,6 +89,22 @@ mod arch_specific {
             eprintln!("WARNING: Missing CPU flags {:?}", missing_cpu_flags);
         }
 
+        Ok(())
+    }
+
+    fn check_kernel_modules(_args: &str) -> Result<()> {
+
+        println!("INFO: check kernel modules: x86_64");
+
+        for module in MOD_LIST.iter() {
+            let _k = check::check_kernel_module_loaded(module);    
+            let res = check_kernel_module_parameters(module);
+
+            // TODO: use (k,res) to generate a Result
+            println!("{res:?}");
+        }
```

Comment:
Right now, the program will panic if check_kernel_module_loaded fails, is that correct?

Reply:
@cmaf: It looks like the intent was to use `check_kernel_module_loaded(module)?`, according to @jodh-intel's comment above:

> @dborquez - When this function is implemented, it would call a generic function that will check if a kernel module is loaded (or if the feature is built into the kernel). Something like:
> 
> check_kernel_module_loaded("kvm")?;
> check_kernel_module_loaded("kvm_intel")?;
> That way, the PR for https://github.com/kata-containers/kata-containers/issues/6132 could also call this generic function. For example:
> 
> check_kernel_module_loaded("vhost_net")?;
> In reality, this generic function will need to accept more than just the module name because we need to also check that module parameters are correct too (for example kvm_intel should be loaded with the unrestricted_guest=Y parameter for that module).
> 
>

Reply:
thanks @cmaf and @c3d , thank you for your inputs. The PR was wip, but it has been updated. 
First step is performed by the `check_kernel_module_loaded` function that verifies the module is loaded.
The second steps is performed by the `check_kernel_module_parameters` function that verifies that the parameters used to load the module are the required.

At the end the function returns a list of text lines consisting of the correct loaded modules and with the incorrect modules / parameters observed.